src/test/javascript/spec/component/participation-submission/participation-submission.spec.ts

```diff
@@ -0,0 +1,110 @@
+    let router: Router;
+    let location: Location;
+
+    beforeEach(async () => {
+        return TestBed.configureTestingModule({
+            imports: [ArtemisTestModule, ArtemisResultModule, ArtemisSharedModule, RouterTestingModule.withRoutes([participationRoute[2]])],
+            declarations: [
+                ParticipationSubmissionComponent,
+                MockComponent(UpdatingResultComponent),
+                MockComponent(TextAssessmentEditorComponent),
+                MockComponent(ResizableInstructionsComponent),
+                MockComponent(TextAssessmentDetailComponent),
+                MockComponent(ComplaintsForTutorComponent),
+            ],
+            providers: [
+                JhiLanguageHelper,
+                { provide: JhiAlertService, useClass: MockAlertService },
+                { provide: AccountService, useClass: MockAccountService },
+                { provide: SessionStorageService, useClass: MockSyncStorage },
+                { provide: LocalStorageService, useClass: MockSyncStorage },
+                { provide: ComplaintService, useClass: MockComplaintService },
+            ],
+        })
+            .compileComponents()
+            .then(() => {
+                fixture = TestBed.createComponent(ParticipationSubmissionComponent);
+                comp = fixture.componentInstance;
+                comp.participationId = 1;
+                debugElement = fixture.debugElement;
+                router = debugElement.injector.get(Router);
+                location = debugElement.injector.get(Location);
+                submissionService = TestBed.get(SubmissionService);
+                findAllSubmissionsOfParticipationStub = stub(submissionService, 'findAllSubmissionsOfParticipation');
+                router.initialNavigation();
+            });
+    });
+
+    afterEach(() => {
+        findAllSubmissionsOfParticipationStub.restore();
+    });
+
+    it('Submissions are correctly loaded from server', fakeAsync(() => {
+        // set all attributes for comp
+        const submissions = [
+            {
+                submissionExerciseType: SubmissionExerciseType.TEXT,
+                id: 2278,
+                submitted: true,
+                type: SubmissionType.MANUAL,
+                submissionDate: moment('2019-07-09T10:47:33.244Z'),
+                text: 'asdfasdfasdfasdf',
+                participation: { id: 1 },
+            },
+        ] as TextSubmission[];
+
+        // check if findAllSubmissionsOfParticipationStub() is called and works
+        findAllSubmissionsOfParticipationStub.returns(of({ body: submissions }));
+        fixture.detectChanges();
+        comp.ngOnInit();
+        tick();
+        expect(findAllSubmissionsOfParticipationStub).to.have.been.called;
+        expect(comp.submissions).to.be.deep.equal(submissions);
+
+        // check if delete button is available
+        const deleteButton = debugElement.query(By.css('#deleteButton'));
```

Comment:
It would be nice if we could also verify the number of rows in the table.

Reply:
I added this: :)
// check if the right amount of rows is visible
const row = debugElement.query(By.css('#row-0'));
expect(row).to.exist;